internal/observability/codes.go

```diff
@@ -47,15 +47,15 @@ func faultFromCode(code yarpcerrors.Code) fault {
 		yarpcerrors.CodeAborted,
 		yarpcerrors.CodeOutOfRange,
 		yarpcerrors.CodeUnauthenticated,
+		yarpcerrors.CodeUnimplemented,
```

Comment:
why would `Unimplemented` be classified as a client error? Given it's generated by the server.

ref: https://grpc.github.io/grpc/core/md_doc_statuscodes.html

Reply:
Being generated by the server and be classifier as a client error is something different.

generated by the server -> what is the origin of the error, client or server. For instance with the error `INTERNAL` and `Error parsing request proto`, this error was generated in the server.
client error -> what is the cause of the error. For instance, `INVALID_ARGUMENT` is due to bad given argument given along the request.

`INVALID_ARGUMENT` is a client error but is always generated by the server

Note: unimplemented can also be generated by the client

Reply:
True, makes sense.  There is definitely some ambiguity here, given Unimplemnted can be returned in many different cases, especially we could also argue that the client has no knowledge on the sever implementation. However, the common cases justifies it better as client error, e.g. espeically grpc libraries will return Unimplmented for a method not found. 

implementation of an endpoint. It seem reasonable to classify it as a client error, but we could also argue that the client has no knowledge on the sever implementation, and it's at the server's control to whether to expose an WIP endpoint or not.

Reply:
Is Unimplemented used when the server receives an API call that doesn't exist? If so, it may be more like a 404 (which are typically because the client requested something that doesn't exist)

Reply:
Based on the documentation:

```
    // Unimplemented indicates operation is not implemented or not
    // supported/enabled in this service.
    //
    // This error code will be generated by the gRPC framework. Most
    // commonly, you will see this error code when a method implementation
    // is missing on the server. It can also be generated for unknown
    // compression algorithms or a disagreement as to whether an RPC should
    // be streaming.
```

so yes this is more like a 404 errors